### ISSUE TITLE
Port/fork Vino + Vino-preferences

### ISSUE THREAD

**Issue by FaberfoX** [mate-desktop/mate-control-center#345]
#### Expected behaviour
Having a Screen Sharing (VNC) server included in MATE

#### Actual behaviour
No Screen Sharing server is included

On 17.10, installing vino and making it autostart was easy enough adding an autostart app (/usr/lib/vino/vino-server --sm-client-disable), and the panel notification triggered vino-preferences, vino was at 3.8 then. 
Vino-preferences was removed after vino 3.8 (https://bugzilla.gnome.org/show_bug.cgi?id=700070) and merged into gnome-control-center. Ubuntu 18.04 includes vino 3.22 so there's no easy way to configure it, the notification is still there but clicking it responds with 
![image](https://user-images.githubusercontent.com/6801510/39414917-6bfb99ba-4c13-11e8-83d6-b89a430a2c36.png)

Is there any chance to have vino-preferences as part of mate-control-center?

**Comment by alexarnaud** [comment]
@FaberfoX Thank you for helping us to make Mate better.

The bug you've reported is not in our responsibility. To make an application visible on the control center, you need to properly edit the desktop file of the application. In a short word: you should ask the maintainer of Vino to make it visible as settings application. 

For example for Compiz Config Manager, the desktop file contains:
> Categories=Settings;DesktopSettings;                                                                                                                                                         

If you need more details, don't hesitate :).

Could it be possible for you to close the bug so?

Best regards,
Alex.

**Comment by FaberfoX** [comment]
@alexarnaud , the issue here is that there's no longer a vino-preferences binary. I installed the pkg from 17.10 (vino_3.8.1-0ubuntu12_amd64.deb) and it works, preferences and all. Anyway, I'll close this as it's not really a bug report, it's a feature request, but before closing it, can you let me know what would be the proper channel to request such a feature?

**Comment by joakim-tjernlund** [comment]
You can make your own vino-preferences from older vino pkg, se my Gentoo ebuild:

`# Copyright 1999-2015 Gentoo Foundation
# Distributed under the terms of the GNU General Public License v2
# $Id$

EAPI="5"
GCONF_DEBUG="yes"
GNOME_ORG_MODULE="vino"

inherit eutils gnome2

DESCRIPTION="An Vino Preferences App"
HOMEPAGE="https://wiki.gnome.org/Projects/Vino"

LICENSE="GPL-2+"
SLOT="0"
KEYWORDS="~alpha amd64 ~arm ~ia64 ~ppc ~ppc64 ~sparc x86 ~x86-fbsd"
IUSE=""

RDEPEND="
	net-misc/vino
"
DEPEND="${RDEPEND}
	>=dev-lang/perl-5
	>=dev-util/intltool-0.50
	virtual/pkgconfig
"

src_configure() {
	gnome2_src_configure \
		--disable-ipv6 \
		--without-gcrypt \
		--without-secret \
		--without-jpeg \
		--without-gnutls \
		--without-telepathy \
		--without-avahi \
		--with-zlib \

}

src_compile() {
	emake vino-preferences
}

src_install() {
	gnome2_src_install
	sed -i "s/OnlyShowIn=.*/OnlyShowIn=Unity;MATE;/" "${D}/etc/xdg/autostart/vino-server.desktop"
	sed -i "s/OnlyShowIn=.*/OnlyShowIn=Unity;MATE;/" "${D}/usr/share/applications/vino-preferences.desktop"
	rm -rf "${D}"/usr/libexec || die "rm libexec failed"
	rm -rf "${D}"/usr/share/dbus-1 || die "rm dbus-1 failed"
	rm -rf "${D}"/usr/share/locale || die "rm locale failed"
	rm -rf "${D}"/usr/share/telepathy || die "rm telepathy failed"
	rm -rf "${D}"/usr/share/GConf || die "rm GConf failed"
	#rm -rf "${D}"/usr/share/glib-2.0 || die "rm glib-2.0 failed"
	cd "${D}"/usr/share/glib-2.0/schemas
	rm org.gnome.Vino.enums.xml || die "rm org.gnome.Vino.enums.xml failed"
	#Need schmema with 'enabled' so rename file and install in parallel to vino
	mv org.gnome.Vino.gschema.xml org.gnome.Vino.Preferences.gschema.xml \
	   || die "mv org.gnome.Vino.gschema.xml failed"
}
`
